app/Platform/Services/PlayerGameActivityService.php

```diff
@@ -280,4 +298,48 @@ public function summarize(): array
             'totalPlaytime' => $totalTime,
         ];
     }
+
+    // returns array of ['agents' => [], 'duration' => 0, 'durationPercentage' => 0.0]
+    public function getClientBreakdown(UserAgentService $userAgentService): array
+    {
+        $clients = [];
+
+        foreach ($this->sessions as $session) {
+            if ($session['userAgent'] ?? null) {
+                $userAgent = $session['userAgent'];
+
+                $decoded = $userAgentService->decode($userAgent);
+                $client = $decoded['client'];
+                if ($decoded['clientVersion'] !== 'Unknown') {
+                    $client .= ' (' . $decoded['clientVersion'] . ')';
+                }
+                if (array_key_exists('clientVariation', $decoded)) {
+                    $client .= ' - ' . $decoded['clientVariation'];
+                }
+
+                if (in_array($client, $clients)) {
+                    $clients[$client]['duration'] = $clients[$client]['duration'] + $session['duration'];
+                    if (!in_array($userAgent, $clients[$client]['agents'])) {
+                        $this->clients[$client]['agents'][] = $userAgent;
```

Comment:
Possible typo and/or logic here? As best as I can tell, `$this->clients` is undefined.